src/test/java/com/juniorstart/juniorstart/controller/ProjectControllerTest.java

```diff
@@ -0,0 +1,182 @@
+import com.juniorstart.juniorstart.repository.TechnologiesRepository;
+import com.juniorstart.juniorstart.repository.UserDao;
+import com.juniorstart.juniorstart.security.UserPrincipal;
+import com.juniorstart.juniorstart.util.MockUtil;
+import org.junit.jupiter.api.*;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.jdbc.EmbeddedDatabaseConnection;
+import org.springframework.boot.test.autoconfigure.jdbc.AutoConfigureTestDatabase;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.data.domain.Pageable;
+import org.springframework.data.domain.Sort;
+import org.springframework.http.MediaType;
+
+import org.springframework.security.test.context.support.WithUserDetails;
+import static org.hamcrest.core.Is.is;
+
+
+
+import java.time.Instant;
+import java.util.*;
+
+
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.*;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.jsonPath;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+public class ProjectControllerTest extends ControllerIntegrationTest {
+
+    private final String BASIC_PROJECT_URL = "/api/project";
+    private final String GET_ALL_URL = BASIC_PROJECT_URL;
+
+
+
+    Project project;
+    Technologies technology;
+    User user;
+    ProjectRequest projectRequest;
+
+    @Autowired
+    ProjectRepository projectRepository;
+
+    @Autowired
+    TechnologiesRepository technologiesRepository;
+
+    @Autowired
+    UserDao userDao;
+
+    @Override
+    @BeforeEach
+    void setUp() throws Exception {
+        super.setUp();
+
+
+        user = User.builder()
+                .name("name")
+                .password("password")
+                .email("some@example.com")
+                .provider(AuthProvider.local)
+                .password("password")
+                .emailVerified(false)
+                .build();
+
+        project = Project.builder()
+                .name("ProjectName")
+                .title("Some title of project")
+                .numberOfSeats(12l)
+                .description("Description of project")
+                .body("Body of project")
+                .recruiting(false)
+                .repository("repository-link")
+                .build();
+
+        projectRequest = ProjectRequest.builder()
+                .title("NewProjecTitle")
+                .repository("NewRepository")
+                .numberOfSeats(12l)
+                .name("SomeName")
+                .description("SomeDescription")
+                .body("SomeBody")
+                .recruiting(false)
+                .build();
+
+
+        userDao.save(user);
+        projectRepository.save(project);
+
+    }
+
+    @AfterEach
+    public void afterAll() {
+        userDao.delete(user);
+        projectRepository.deleteAll();
+    }
```

Comment:
@DirtiesContext(classMode = DirtiesContext.ClassMode.AFTER_EACH_TEST_METHOD) to w abstrakcyjnej zamiast public void afterAll()